managed-ledger/src/main/java/org/apache/bookkeeper/mledger/LedgerOffloader.java

```diff
@@ -91,6 +90,10 @@ public OffloadResult(long beginLedger, long beginEntry, long endLedger, long end
     String METADATA_SOFTWARE_VERSION_KEY = "S3ManagedLedgerOffloaderSoftwareVersion";
     String METADATA_SOFTWARE_GITSHA_KEY = "S3ManagedLedgerOffloaderSoftwareGitSha";
 
+    default LedgerOffloader fork() {
+        throw new UnsupportedOperationException(this.getClass().toString() + " fork()");
+    }
```

Comment:
This method will create a new instance with the same configuration as the current one. We need it because the life cycle of `ManagedledgerImpl.config.getLedgerOffloader()` is not stable, for streaming offload we need to have control of the offloader instance. I will add some description.